embassy-rp/src/clocks.rs

```diff
@@ -624,12 +626,12 @@ pub fn clk_rtc_freq() -> u16 {
     CLOCKS.rtc.load(Ordering::Relaxed)
 }
 
-fn start_xosc(crystal_hz: u32) {
+fn start_xosc(crystal_hz: u32, delay_multiplier: u32) {
     pac::XOSC
         .ctrl()
         .write(|w| w.set_freq_range(pac::xosc::vals::CtrlFreqRange::_1_15MHZ));
 
-    let startup_delay = ((crystal_hz / 1000) + 128) / 256;
+    let startup_delay = (((crystal_hz / 1000) + 128) * delay_multiplier) / 256;
```

Comment:
The `(x + 128)/256` is to divide by 256 rounding to nearest. I'd try to keep it: `((crystal_hz / 1000 + delay_multiplier) + 128) / 256`

Reply:
Check f69e845 for improvement

Reply:
oops sorry! I wrote `+ delay_multiplier` but it should be `* delay_multiplier`